brototype.js

```diff
@@ -68,18 +68,19 @@
             if (Array.isArray(key)) {
                 var index, value, result = [];
                 for (index in key) {
-                    if (value = this.iCanHaz(key[index])) {
-                        result.push(value);
-                    }
+                    value = this.iCanHaz(key[index]);
+                    result.push(value);
                 }
                 return result;
             }
             var props = key.split('.'),
                 item = this.obj;
-            for (var i = 0; i < props.length; i++) {
-                item = item[props[i]];
-                if (Bro(item).isThatEvenAThing() === Bro.NOWAY) {
-                    return item;
+            if (item !== undefined) {
```

Comment:
Not trying to be a dick here, but I really think it needs to use `typeof`...

If I open the Chrome console and type `foo === undefined`, I get an exception, whereas if I type `typeof foo`, I get "undefined".

I think to avoid any exceptions in strict browsers, it really needs to be `typeof item !== 'undefined'`.